ingress/src/main/java/com/redhat/developer/ingress/producer/KafkaEventPublisher.java

```diff
@@ -21,16 +23,27 @@
 
     private final BroadcastProcessor<String> eventSubject = BroadcastProcessor.create();
 
-    public void sendEvent(CloudEvent cloudEvent) {
-        LOGGER.info("[ingress] Sending cloudEvent with id '{}' to event queue", cloudEvent.getId());
+    /*
+     * Add our specific metadata to the incoming event
+     */
+    private CloudEvent addMetadataToIncomingEvent(String bridgeId, CloudEvent cloudEvent) {
+        return CloudEventBuilder.v1(cloudEvent)
+                .withExtension(CloudEventExtensions.BRIDGE_ID_EXTENSION, bridgeId)
+                .build();
+    }
```

Comment:
I was assuming we'd have a set of reserved extensions on the CloudEvent that it would be invalid to include when you send an Event to the Bridge. If we're comfortable with that I can add the initial validation to the Ingress to validate that the user is not attempting to use our reserved stuff....

This of course does put is in the position where the user is for whatever reason completely tied to a particular set of attributes on their CloudEvent that clash with our reserved ones, they cannot send an Event to the Bridge. I think the chances of that are low (not zero of course) but it may be something that we'd want to enforce anyway, just to make sure we don't end up in a situation where we think things we added to the CloudEvent by our internal logic, but were in fact not..

The alternative is, as you'd previously suggested, to wrap the received CloudEvent into an internal Event. I'm still not 100% on wrapping it right now over enforcing a reserved set of  extensions as see my comment above about potentially wanting to enforce their non-usage anyway.

RE: the Event clean-up. 100% agree with you if the Event is being sent out of the Bridge "as-is". In other circumstances, I wasn't sure yet how that would look until we'd handled Filters, Transformations and Actions. I'll add a TODO to the code to make sure we don't forget to do it.

Reply:
It's fine to have a reserved extension, the important thing is that if the user tries to use them we reject the message or we throw an exception. 
We have just to be careful about malicious users. For example, we have to prevent the user to specify that extension with a `bridgeid` that does not belong to him and do bad stuff :)

Reply:
Captured as: https://issues.redhat.com/browse/MGDOBR-35